unsuccessful "npm install"

Hi,
I’d like to play with mutt-hue-bridge in my setup ( rasbPI 2, esp8266/nodemcu etc. etc. ) but unfortunately I was not able to install it ( maybe my noob level about topic like node.js didn’t help :-( ).

In attach you’ll find the "npm install” result and below a short  description about my environment  :
- raspi 2 / jarvis 
- node.js 4.3.1 / npm 2.14.12
- mosquitto 1.4.8

Could you help me?

[mqtt_err.txt](https://github.com/dennisdegreef/mqtt-hue-bridge/files/158958/mqtt_err.txt)

Sorry, the file … :-)

> Il giorno 04/mar/2016, alle ore 18:40, GMail massimo.romagnoli@gmail.com ha scritto:
> 
> Wow … thanks for your superfast reply !!! :-)
> 
> The RaspPI OS is Raspbian Jessie ( sorry I didn’t know why I named it “jarvis" in my previous msg. :-( ) and here attached the “english” version of the "npm install” outcome.
> 
> Thanks for your support 
> 
> Ciao
> 
> max
> 
> > Il giorno 04/mar/2016, alle ore 17:27, Dennis de Greef <notifications@github.com <mailto:notifications@github.com>> ha scritto:
> > 
> > Hi iomax,
> > 
> > Awesome that you have a similar setup!
> > 
> > I just tried to do a fresh npm install, which works fine.
> > Although my node/npm versions on this machine are a bit behind,
> > I can check on another machine this weekend.
> > 
> > Could you try to set your locale to english for the logs, temporarily?
> > What operating system are you running?
> > 
> > I'm by no means an expert in node.js, these projects we're my first in Node, but i'd be glad to think with you on this issue.
> > 
> > —
> > Reply to this email directly or view it on GitHub https://github.com/dennisdegreef/mqtt-hue-bridge/issues/2#issuecomment-192343253.

osmc@c1p8:~$ cd mqtt-hue-bridge/
osmc@c1p8:~/mqtt-hue-bridge$ npm install

> bufferutil@1.1.0 install /home/osmc/mqtt-hue-bridge/node_modules/mqtt/node_modules/websocket-stream/node_modules/ws/node_modules/bufferutil
> node-gyp rebuild

make: ingresso nella directory "/home/osmc/mqtt-hue-bridge/node_modules/mqtt/node_modules/websocket-stream/node_modules/ws/node_modules/bufferutil/build"
  CXX(target) Release/obj.target/bufferutil/src/bufferutil.o
In file included from ../src/bufferutil.cc:16:0:
../node_modules/nan/nan.h:261:25: error: redefinition of ‘template<class T> v8::Local<T> _NanEnsureLocal(v8::Local<T>)’
 NAN_INLINE v8::Local<T> _NanEnsureLocal(v8::Local<T> val) {
                         ^
../node_modules/nan/nan.h:256:25: note: ‘template<class T> v8::Local<T> _NanEnsureLocal(v8::Handle<T>)’ previously declared here
 NAN_INLINE v8::Local<T> _NanEnsureLocal(v8::Handle<T> val) {
                         ^
../node_modules/nan/nan.h:661:13: error: ‘node::smalloc’ has not been declared
     , node::smalloc::FreeCallback callback
             ^
../node_modules/nan/nan.h:661:35: error: expected ‘,’ or ‘...’ before ‘callback’
     , node::smalloc::FreeCallback callback
                                   ^
../node_modules/nan/nan.h: In function ‘v8::Localv8::Object NanNewBufferHandle(char_, size_t, int)’:
../node_modules/nan/nan.h:665:50: error: ‘callback’ was not declared in this scope
         v8::Isolate::GetCurrent(), data, length, callback, hint);
                                                  ^
../node_modules/nan/nan.h:665:60: error: ‘hint’ was not declared in this scope
         v8::Isolate::GetCurrent(), data, length, callback, hint);
                                                            ^
../node_modules/nan/nan.h: In function ‘v8::Localv8::Object NanNewBufferHandle(const char_, uint32_t)’:
../node_modules/nan/nan.h:672:67: error: call of overloaded ‘New(v8::Isolate_, const char_&, uint32_t&)’ is ambiguous
     return node::Buffer::New(v8::Isolate::GetCurrent(), data, size);
                                                                   ^
../node_modules/nan/nan.h:672:67: note: candidates are:
In file included from ../src/bufferutil.cc:10:0:
/home/osmc/.node-gyp/4.3.1/include/node/node_buffer.h:31:40: note: v8::MaybeLocalv8::Object node::Buffer::New(v8::Isolate_, v8::Localv8::String, node::encoding) <near match>
 NODE_EXTERN v8::MaybeLocalv8::Object New(v8::Isolate_ isolate,
                                        ^
/home/osmc/.node-gyp/4.3.1/include/node/node_buffer.h:31:40: note:   no known conversion for argument 3 from ‘uint32_t {aka unsigned int}’ to ‘node::encoding’
/home/osmc/.node-gyp/4.3.1/include/node/node_buffer.h:43:40: note: v8::MaybeLocalv8::Object node::Buffer::New(v8::Isolate_, char_, size_t) <near match>
 NODE_EXTERN v8::MaybeLocalv8::Object New(v8::Isolate\* isolate,
                                        ^
/home/osmc/.node-gyp/4.3.1/include/node/node_buffer.h:43:40: note:   no known conversion for argument 2 from ‘const char_’ to ‘char_’
In file included from ../src/bufferutil.cc:16:0:
../node_modules/nan/nan.h: In function ‘v8::Localv8::Object NanNewBufferHandle(uint32_t)’:
../node_modules/nan/nan.h:676:61: error: could not convert ‘node::Buffer::New(v8::Isolate::GetCurrent(), size)’ from ‘v8::MaybeLocalv8::Object’ to ‘v8::Localv8::Object’
     return node::Buffer::New(v8::Isolate::GetCurrent(), size);
                                                             ^
../node_modules/nan/nan.h: In function ‘v8::Localv8::Object NanBufferUse(char_, uint32_t)’:
../node_modules/nan/nan.h:683:12: error: ‘Use’ is not a member of ‘node::Buffer’
     return node::Buffer::Use(v8::Isolate::GetCurrent(), data, size);
            ^
bufferutil.target.mk:90: set di istruzioni per l'obiettivo "Release/obj.target/bufferutil/src/bufferutil.o" non riuscito
make: *_\* [Release/obj.target/bufferutil/src/bufferutil.o] Errore 1
make: uscita dalla directory "/home/osmc/mqtt-hue-bridge/node_modules/mqtt/node_modules/websocket-stream/node_modules/ws/node_modules/bufferutil/build"
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Linux 4.4.0-1-osmc
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/osmc/mqtt-hue-bridge/node_modules/mqtt/node_modules/websocket-stream/node_modules/ws/node_modules/bufferutil
gyp ERR! node -v v4.3.1
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok 

> utf-8-validate@1.1.0 install /home/osmc/mqtt-hue-bridge/node_modules/mqtt/node_modules/websocket-stream/node_modules/ws/node_modules/utf-8-validate
> node-gyp rebuild

make: ingresso nella directory "/home/osmc/mqtt-hue-bridge/node_modules/mqtt/node_modules/websocket-stream/node_modules/ws/node_modules/utf-8-validate/build"
  CXX(target) Release/obj.target/validation/src/validation.o
In file included from ../src/validation.cc:15:0:
../node_modules/nan/nan.h:261:25: error: redefinition of ‘template<class T> v8::Local<T> _NanEnsureLocal(v8::Local<T>)’
 NAN_INLINE v8::Local<T> _NanEnsureLocal(v8::Local<T> val) {
                         ^
../node_modules/nan/nan.h:256:25: note: ‘template<class T> v8::Local<T> _NanEnsureLocal(v8::Handle<T>)’ previously declared here
 NAN_INLINE v8::Local<T> _NanEnsureLocal(v8::Handle<T> val) {
                         ^
../node_modules/nan/nan.h:661:13: error: ‘node::smalloc’ has not been declared
     , node::smalloc::FreeCallback callback
             ^
../node_modules/nan/nan.h:661:35: error: expected ‘,’ or ‘...’ before ‘callback’
     , node::smalloc::FreeCallback callback
                                   ^
../node_modules/nan/nan.h: In function ‘v8::Localv8::Object NanNewBufferHandle(char_, size_t, int)’:
../node_modules/nan/nan.h:665:50: error: ‘callback’ was not declared in this scope
         v8::Isolate::GetCurrent(), data, length, callback, hint);
                                                  ^
../node_modules/nan/nan.h:665:60: error: ‘hint’ was not declared in this scope
         v8::Isolate::GetCurrent(), data, length, callback, hint);
                                                            ^
../node_modules/nan/nan.h: In function ‘v8::Localv8::Object NanNewBufferHandle(const char_, uint32_t)’:
../node_modules/nan/nan.h:672:67: error: call of overloaded ‘New(v8::Isolate_, const char_&, uint32_t&)’ is ambiguous
     return node::Buffer::New(v8::Isolate::GetCurrent(), data, size);
                                                                   ^
../node_modules/nan/nan.h:672:67: note: candidates are:
In file included from ../src/validation.cc:10:0:
/home/osmc/.node-gyp/4.3.1/include/node/node_buffer.h:31:40: note: v8::MaybeLocalv8::Object node::Buffer::New(v8::Isolate_, v8::Localv8::String, node::encoding) <near match>
 NODE_EXTERN v8::MaybeLocalv8::Object New(v8::Isolate_ isolate,
                                        ^
/home/osmc/.node-gyp/4.3.1/include/node/node_buffer.h:31:40: note:   no known conversion for argument 3 from ‘uint32_t {aka unsigned int}’ to ‘node::encoding’
/home/osmc/.node-gyp/4.3.1/include/node/node_buffer.h:43:40: note: v8::MaybeLocalv8::Object node::Buffer::New(v8::Isolate_, char_, size_t) <near match>
 NODE_EXTERN v8::MaybeLocalv8::Object New(v8::Isolate\* isolate,
                                        ^
/home/osmc/.node-gyp/4.3.1/include/node/node_buffer.h:43:40: note:   no known conversion for argument 2 from ‘const char_’ to ‘char_’
In file included from ../src/validation.cc:15:0:
../node_modules/nan/nan.h: In function ‘v8::Localv8::Object NanNewBufferHandle(uint32_t)’:
../node_modules/nan/nan.h:676:61: error: could not convert ‘node::Buffer::New(v8::Isolate::GetCurrent(), size)’ from ‘v8::MaybeLocalv8::Object’ to ‘v8::Localv8::Object’
     return node::Buffer::New(v8::Isolate::GetCurrent(), size);
                                                             ^
../node_modules/nan/nan.h: In function ‘v8::Localv8::Object NanBufferUse(char_, uint32_t)’:
../node_modules/nan/nan.h:683:12: error: ‘Use’ is not a member of ‘node::Buffer’
     return node::Buffer::Use(v8::Isolate::GetCurrent(), data, size);
            ^
validation.target.mk:90: set di istruzioni per l'obiettivo "Release/obj.target/validation/src/validation.o" non riuscito
make: *_\* [Release/obj.target/validation/src/validation.o] Errore 1
make: uscita dalla directory "/home/osmc/mqtt-hue-bridge/node_modules/mqtt/node_modules/websocket-stream/node_modules/ws/node_modules/utf-8-validate/build"
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Linux 4.4.0-1-osmc
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/osmc/mqtt-hue-bridge/node_modules/mqtt/node_modules/websocket-stream/node_modules/ws/node_modules/utf-8-validate
gyp ERR! node -v v4.3.1
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok 
npm WARN optional dep failed, continuing bufferutil@1.1.0
npm WARN optional dep failed, continuing utf-8-validate@1.1.0
node-hue-api@1.0.5 node_modules/node-hue-api
├── parseUri@1.2.3-2
├── traits@0.4.0
├── q@1.0.1
├── xml2js@0.2.8 (sax@0.5.8)
└── request-util@0.1.0 (request@2.51.0)

mqtt@1.1.3 node_modules/mqtt
├── inherits@2.0.1
├── xtend@4.0.1
├── minimist@1.2.0
├── commist@1.0.0 (leven@1.0.2)
├── readable-stream@1.0.33 (string_decoder@0.10.31, isarray@0.0.1, core-util-is@1.0.2)
├── mqtt-connection@2.1.1 (through2@0.6.5, reduplexer@1.1.0)
├── mqtt-packet@3.4.6 (bl@0.9.5)
├── end-of-stream@1.1.0 (once@1.3.3)
├── help-me@0.1.0 (pump@1.0.1)
├── concat-stream@1.5.1 (typedarray@0.0.6, readable-stream@2.0.5)
└── websocket-stream@1.5.2 (through2@0.6.5, duplexify@3.4.3, ws@0.7.2)
osmc@c1p8:~/mqtt-hue-bridge$

--- a/package.json
+++ b/package.json
@@ -5,8 +5,8 @@
     "license": "MIT",
     "version": "1.0.0",
     "dependencies": {
-        "mqtt": "1.1.3",
-        "node-hue-api": "1.0.5"
+        "mqtt": "^1.1",
+        "node-hue-api": "^1.0"
     },
     "repository" : {
         "type" : "git",

